### PR TITLE
feature(alias): allow loader query to be used for aliased loader

### LINKED ISSUE #3320: resolveLoader.alias does not handle loader query properly.
**I'm submitting a bug report**

**Webpack version:**
2.1.0-beta.27 

**Please tell us about your environment:**
Linux

**Current behavior:**
Trying to alias from `loader` to `loader?query` broken

**Expected/desired behavior:**
If i `require('loader!resource')`, when `loader` is aliased to `loader?query`, then require should be transformed to `require('loader?query!resource')`. As for now, webpack tries to resolve `loader?query` as path to js file with loader.

It is working in webpack 1.x branch. Is there any workarounds available?

### PR DESCRIPTION
**Pull request by TheLarkInn**
**What kind of change does this PR introduce?**
Feature/Bugfixish

**Did you add tests for your changes?**
Yes

**If relevant, link to documentation update:**
N/A

**Summary**

Fixes #3320

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: test/configCases/loaders/issue-3320/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+it("should resolve aliased loader module with query (issue-3320)", function() {
+	var foo = require('./a');
```
Comment (sokra) [inline test/configCases/loaders/issue-3320/index.js]: It should validate the returned value.

Comment (TheLarkInn) [inline test/configCases/loaders/issue-3320/index.js]: Yeah just spotted that. I forgot to save changes. Update coming shortly.

Comment (TheLarkInn) [inline test/configCases/loaders/issue-3320/index.js]: done

File: lib/NormalModuleFactory.js [hunk 2]
```diff
@@ -239,9 +240,20 @@ NormalModuleFactory.prototype.resolveRequestArray = function resolveRequestArray
 				});
 			}
 			if(err) return callback(err);
-			return callback(null, objectAssign({}, item, {
+
+			if(result.indexOf("?") > -1) {
+				resultQuery = result.substr(result.indexOf("?") + 1, result.length);
```
Comment (TheLarkInn) [inline lib/NormalModuleFactory.js]: Done

**Comment by TheLarkInn** [comment]
Once webpack/enhanced-resolve#51 is merged, this should pass. (From my local linked-package testing).

File: test/configCases/loaders/issue-3320/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+			"any-loader": 'any-loader?foo="someMessage"'
+		}
+	},
+	module: {
+		rules: [
+			{
+				test: /a\.js$/,
+				use: [
+					{
+						loader: "any-loader"
```
Comment (sokra) [inline test/configCases/loaders/issue-3320/webpack.config.js]: Could you add a test which uses `any-loader?foo="abc"` in the rules?

Comment (TheLarkInn) [inline test/configCases/loaders/issue-3320/webpack.config.js]: Yup sounds good.

**Comment by sokra** [comment]
Thanks

**Pull request merged**